### ISSUE TITLE
New version toolbox icon doesn't show up on toolbar

### ISSUE THREAD

**Issue by kiwichick888** [ffes/nppsnippets#51]
In the new version 'Show Toolbox Icon' doesn't appear to be working. The button doesn't appear on the toolbar. I've tried with a clean install of Npp but it made no difference.

**Comment by ffes** [comment]
Someone needs to design/make dark and light icon.

And know that it is just the icon in the toolbar. You can always use the menu item to show or hide the snippet window.

**Comment by ipro-hcqip** [comment]
How do you create custom toolbar button?

**Comment by kiwichick888** [comment]
> How do you create custom toolbar button?

Using the Customise Toolbar plugin. And you need to source an image for the toolbar button.

**Comment by vico93** [comment]
> > And know that it is just the icon in the toolbar. You can always use the menu item to show or hide the snippet window.
> 
> Thanks, yes, I know this, and for now I've created a custom toolbar button but it would be nice if it worked as it should.
...
> Using the Customize Toolbar plugin. And you need to source an image for the toolbar button.

What do you mean by "create a custom toolbar button" using Customize Toolbar?

I was going to say that after the mentioned update the NppSnippets icon also disappeared from the available icons on the Customize Toolbar menu:
![image](https://user-images.githubusercontent.com/3485505/212692197-83207a44-795c-4201-82f8-cb7b4f8a2b16.png)
It always sat on the bottom of the list on the left (if not active on the toolbar), but nowdays it completely dissapeared.

**Comment by vico93** [comment]
> And know that it is just the icon in the toolbar. You can always use the menu item to show or hide the snippet window.

But for 'UX ergonomics' purposes, it's always great to have a easily accessible button to quickly toggle the visibility of the snippets column...

**Comment by kiwichick888** [comment]
> What do you mean by "create a custom toolbar button" using Customize Toolbar?

Customize Toolbar comes with a default set of toolbar buttons but you can also create your own. In Npp, go to the Plugins menu > Customize Toolbar > Help - Custom Buttons. That explains how to create them.

**Comment by sykledust** [comment]
It seems this is still an issue, right?

**Comment by kiwichick888** [comment]
> It seems this is still an issue, right?

Yes, it is. Very disappointing that nothing has been done about it, especially as it has been tagged as a bug.

**Comment by Fwgd** [comment]
Walk trough for the solution with the ‘Customize Toolbar’ plugin:
- In the menu bar, activate⠀Plugins > Customize Toolbar > ‘Custom Buttons’
- Restart N++, the configuration file⠀‘CustomizeToolbar.btn’⠀is created in folder ...\plugins\Config
- Edit this file and insert this line:
`Plugins,Snippets,Snippets,,snippets-toolbar-icon.bmp`
- Copy attached image⠀‘snippets-toolbar-icon.bmp’⠀into folder ...\plugins\Config.⠀(image source: ‘NppSnippets.dll’)
[snippets-toolbar-icon.bmp](https://github.com/user-attachments/files/23219058/snippets-toolbar-icon.bmp)

**Comment by kiwichick888** [comment]
> Walk trough for the solution with the ‘Customize Toolbar’ plugin:
> 
>     * In the menu bar, activate⠀Plugins > Customize Toolbar > ‘Custom Buttons’
> 
>     * Restart N++, the configuration file⠀‘CustomizeToolbar.btn’⠀is created in folder ...\plugins\Config
> 
>     * Edit this file and insert this line:
>       `Plugins,Snippets,Snippets,,snippets-toolbar-icon.bmp`
> 
>     * Copy attached image⠀‘snippets-toolbar-icon.bmp’⠀into folder ...\plugins\Config.⠀(image source: ‘NppSnippets.dll’)
>       [snippets-toolbar-icon.bmp](https://github.com/user-attachments/files/23219058/snippets-toolbar-icon.bmp)

I've tried doing this but bmp images don't seem to work for me. I have standard icons selected in Npp preferences and the bmp you supplied seems to be formatted correctly.

<img width="1920" height="1023" alt="Image" src="https://github.com/user-attachments/assets/6cf55977-2282-4bcc-a125-857e0bf00c3e" />

**Comment by Fwgd** [comment]
> I've tried doing this but bmp images don't seem to work for me. I have standard icons selected in Npp preferences and the bmp you supplied seems to be formatted correctly.

I scaled the image from the ‘NppSnippets.dll’ (16x16px) to 32x32px (32 bit) and saved it as an ICO, maybe that will work.
See [snippets-toolbar-icon.zip](https://github.com/user-attachments/files/24280848/snippets-toolbar-icon.zip).
Edited code section:
`Plugins,Snippets,Snippets,,snippets-toolbar-icon.bmp,snippets-toolbar-icon.ico`

**Comment by kiwichick888** [comment]
> I scaled the image from the ‘NppSnippets.dll’ (16x16px) to 32x32px (32 bit) and saved it as an ICO, maybe that will work. See [snippets-toolbar-icon.zip](https://github.com/user-attachments/files/24280848/snippets-toolbar-icon.zip). Edited code section: `Plugins,Snippets,Snippets,,snippets-toolbar-icon.bmp,snippets-toolbar-icon.ico`

Yes, I know I can do that but it doesn't change the fact that bmp images don't work.
